what4-abc and what4-blt fail to compile due to SolverEndSATQuery changes

Attempting to run `cabal build all` on `master` will yield the following build errors on `what4-abc`:

<details>

```
[1 of 1] Compiling What4.Solver.ABC ( src/What4/Solver/ABC.hs, /home/rscott/Documents/Hacking/Haskell/what4-2/dist-newstyle/build/x86_64-linux/ghc-8.10.4/what4-abc-0.1/build/What4/Solver/ABC.o, /home/rscott/Documents/Hacking/Haskell/what4-2/dist-newstyle/build/x86_64-linux/ghc-8.10.4/what4-abc-0.1/build/What4/Solver/ABC.dyn_o )

src/What4/Solver/ABC.hs:747:7: error:
    Not in scope: ‘satQuerySolverName’
    Perhaps you want to add ‘satQuerySolverName’ to the import list
    in the import of ‘What4.Interface’
    (src/What4/Solver/ABC.hs:(81,1)-(82,95)).
    |
747 |     { satQuerySolverName = "ABC"
    |       ^^^^^^^^^^^^^^^^^^

src/What4/Solver/ABC.hs:748:7: error:
    Not in scope: ‘satQueryReason’
    Perhaps you want to add ‘satQueryReason’ to the import list
    in the import of ‘What4.Interface’
    (src/What4/Solver/ABC.hs:(81,1)-(82,95)).
    |
748 |     , satQueryReason = logReason logData
    |       ^^^^^^^^^^^^^^

src/What4/Solver/ABC.hs:794:9: error:
    Not in scope: ‘satQueryResult’
    Perhaps you want to add ‘satQueryResult’ to the import list
    in the import of ‘What4.Interface’
    (src/What4/Solver/ABC.hs:(81,1)-(82,95)).
    |
794 |       { satQueryResult = forgetModelAndCore res
    |         ^^^^^^^^^^^^^^

src/What4/Solver/ABC.hs:795:9: error:
    Not in scope: ‘satQueryError’
    Perhaps you want to add ‘satQueryError’ to the import list
    in the import of ‘What4.Interface’
    (src/What4/Solver/ABC.hs:(81,1)-(82,95)).
    |
795 |       , satQueryError  = Nothing
    |         ^^^^^^^^^^^^^
```
</details>

And on `what4-blt`:

<details>

```
[1 of 1] Compiling What4.Solver.BLT ( src/What4/Solver/BLT.hs, /home/rscott/Documents/Hacking/Haskell/what4-2/dist-newstyle/build/x86_64-linux/ghc-8.10.4/what4-blt-0.2/build/What4/Solver/BLT.o, /home/rscott/Documents/Hacking/Haskell/what4-2/dist-newstyle/build/x86_64-linux/ghc-8.10.4/what4-blt-0.2/build/What4/Solver/BLT.dyn_o )

src/What4/Solver/BLT.hs:135:5: error:
    • Constructor ‘SolverStartSATQuery’ does not have field ‘satQuerySolverName’
    • In the second argument of ‘logSolverEvent’, namely
        ‘SolverStartSATQuery
           {satQuerySolverName = "BLT", satQueryReason = logReason logData}’
      In a stmt of a 'do' block:
        logSolverEvent
          sym
          SolverStartSATQuery
            {satQuerySolverName = "BLT", satQueryReason = logReason logData}
      In the expression:
        do let cfg = getConfiguration sym
           epar <- parseBLTParams . T.unpack
                     <$> (getOpt =<< getOptionSetting bltParams cfg)
           par <- either fail return epar
           logSolverEvent
             sym
             SolverStartSATQuery
               {satQuerySolverName = "BLT", satQueryReason = logReason logData}
           ....
    |
135 |     SolverStartSATQuery
    |     ^^^^^^^^^^^^^^^^^^^...

src/What4/Solver/BLT.hs:135:5: error:
    • Constructor ‘SolverStartSATQuery’ does not have field ‘satQueryReason’
    • In the second argument of ‘logSolverEvent’, namely
        ‘SolverStartSATQuery
           {satQuerySolverName = "BLT", satQueryReason = logReason logData}’
      In a stmt of a 'do' block:
        logSolverEvent
          sym
          SolverStartSATQuery
            {satQuerySolverName = "BLT", satQueryReason = logReason logData}
      In the expression:
        do let cfg = getConfiguration sym
           epar <- parseBLTParams . T.unpack
                     <$> (getOpt =<< getOptionSetting bltParams cfg)
           par <- either fail return epar
           logSolverEvent
             sym
             SolverStartSATQuery
               {satQuerySolverName = "BLT", satQueryReason = logReason logData}
           ....
    |
135 |     SolverStartSATQuery
    |     ^^^^^^^^^^^^^^^^^^^...

src/What4/Solver/BLT.hs:143:7: error:
    • Constructor ‘SolverEndSATQuery’ does not have field ‘satQueryResult’
    • In the second argument of ‘logSolverEvent’, namely
        ‘SolverEndSATQuery
           {satQueryResult = forgetModelAndCore result,
            satQueryError = Nothing}’
      In a stmt of a 'do' block:
        logSolverEvent
          sym
          SolverEndSATQuery
            {satQueryResult = forgetModelAndCore result,
             satQueryError = Nothing}
      In the expression:
        do forM_ ps (assume h)
           result <- checkSat h
           logSolverEvent
             sym
             SolverEndSATQuery
               {satQueryResult = forgetModelAndCore result,
                satQueryError = Nothing}
           contFn result
    |
143 |       SolverEndSATQuery
    |       ^^^^^^^^^^^^^^^^^...

src/What4/Solver/BLT.hs:143:7: error:
    • Constructor ‘SolverEndSATQuery’ does not have field ‘satQueryError’
    • In the second argument of ‘logSolverEvent’, namely
        ‘SolverEndSATQuery
           {satQueryResult = forgetModelAndCore result,
            satQueryError = Nothing}’
      In a stmt of a 'do' block:
        logSolverEvent
          sym
          SolverEndSATQuery
            {satQueryResult = forgetModelAndCore result,
             satQueryError = Nothing}
      In the expression:
        do forM_ ps (assume h)
           result <- checkSat h
           logSolverEvent
             sym
             SolverEndSATQuery
               {satQueryResult = forgetModelAndCore result,
                satQueryError = Nothing}
           contFn result
    |
143 |       SolverEndSATQuery
    |       ^^^^^^^^^^^^^^^^^...
```
</details>

This is due to #111. Patch incoming.

--- a/what4-abc/src/What4/Solver/ABC.hs
+++ b/what4-abc/src/What4/Solver/ABC.hs
@@ -79,7 +79,8 @@ import           What4.BaseTypes
 import           What4.Concrete
 import           What4.Config
 import           What4.Interface
-                   (getConfiguration, IsExprBuilder, logSolverEvent, SolverEvent(..), andAllOf)
+                   ( getConfiguration, IsExprBuilder, logSolverEvent
+                   , SolverEvent(..), SolverStartSATQuery(..), SolverEndSATQuery(..), andAllOf )
 import           What4.Expr
 import           What4.Expr.Builder
 import qualified What4.Expr.BoolMap as BM
@@ -743,10 +744,10 @@ checkSat sym logData e = do
   checkSupportedByAbc vars
   checkNoLatches vars
   logSolverEvent sym
-    SolverStartSATQuery
+    (SolverStartSATQuery $ SolverStartSATQueryRec
     { satQuerySolverName = "ABC"
     , satQueryReason = logReason logData
-    }
+    })
   withNetwork $ \ntk -> do
     -- Get network
     let g = gia ntk
@@ -790,10 +791,10 @@ checkSat sym logData e = do
              logCallbackVerbose logData 2 "Calling ABC's QBF solver"
              runQBF ntk exist_cnt p max_qbf_iter
     logSolverEvent sym
-      SolverEndSATQuery
+      (SolverEndSATQuery $ SolverEndSATQueryRec
       { satQueryResult = forgetModelAndCore res
       , satQueryError  = Nothing
-      }
+      })
     return res
 
 -- | Associate an element in a binding with the term.

--- a/what4-blt/src/What4/Solver/BLT.hs
+++ b/what4-blt/src/What4/Solver/BLT.hs
@@ -132,18 +132,18 @@ runBLTInOverride sym logData ps contFn = do
   epar <- parseBLTParams . T.unpack <$> (getOpt =<< getOptionSetting bltParams cfg)
   par  <- either fail return epar
   logSolverEvent sym
-    SolverStartSATQuery
+    (SolverStartSATQuery $ SolverStartSATQueryRec
     { satQuerySolverName = "BLT"
     , satQueryReason = logReason logData
-    }
+    })
   withHandle par $ \h -> do
     forM_ ps (assume h)
     result <- checkSat h
     logSolverEvent sym
-      SolverEndSATQuery
+      (SolverEndSATQuery $ SolverEndSATQueryRec
       { satQueryResult = forgetModelAndCore result
       , satQueryError  = Nothing
-      }
+      })
     contFn result
 
 ------------------------------------------------------------------------

